plutus-core/plutus-core/src/Universe/Core.hs

```diff
@@ -595,6 +609,14 @@ withApplicable _ _ k =
             withTypeable repB k
         _ -> mzero
 
+-- | Apply a type constructor to an argument, provided kinds match.
+tryUniApply
+    :: (MonadPlus m, HasUniApply uni)
```

Comment:
both here and in `withTypeable`, it seems like you maybe just want `Maybe` rather than `MonadPlus m`?